### ISSUE TITLE
epoxy_glx_version hands NULL to sscanf if no GLX active on display

### ISSUE THREAD

**Issue by ssbertilson** [anholt/libepoxy#72]
```
version_string = glXQueryServerString(dpy, screen, GLX_VERSION);
ret = sscanf(version_string, "%d.%d", &server_major, &server_minor);
```

Was presented with libc segfault called by lightdm-gtk-greeter as a result of inability to modload fglrx due to pci_ignore_hotplug being exported GPL.

Maybe there should be an assert here for:
version_string != NULL

**Comment by yaronct** [comment]
@ssbertilson: This repository doesn't seem active lately, but I'v fixed it in [my fork](https://github.com/yaronct/libepoxy), if you're interested. Thanx for pointing out this problem.

**Comment by Claghorn** [comment]
It is 2016 now, and this crops up in fedora24. I had to build from source and fix several places to check for NULL pointers to be able to run some gtk3 programs under x2go (which has no GL).

See: https://bugzilla.redhat.com/show_bug.cgi?id=1395366

**Comment by yaronct** [comment]
@Claghorn: Have u tried branch `master`?

**Comment by Claghorn** [comment]
I'm more concerned that whatever version is released in fedora still has the problem than which branch it might be fixed on. It was simple enough to add my own fixes. I have no idea where fedora is getting the one they build (and there hasn't been any response to my fedora bug report, so I'm not sure anyone is paying any attention).

**Comment by yaronct** [comment]
@Claghorn: I'm sorry that the bug still persists in the latest Epoxy stable version. Epoxy has indeed not been maintained for quite some time due to @anholt's lack of time. I've been recently become collaborator in this repository and do my best, with my limited time, to go over the issues and pull requests and make a new release. Then, gradually, distributions will update their packages and bugs will b fixed in the distributions.

I do ask for your help, though, to test whether the bug persists in branch `master` - otherwise we must fix it b4 making the new release. I'm sorry for the inconvenience u and others have been experiencing.

**Comment by Claghorn** [comment]
I don't think I need to build it, just glancing at the source shows it getting version_string from a GL query and passing it directly to sscanf with no NULL check. That may well be the version fedora is using (master seems like a likely branch for them to pick up). If so, the patch in my fedora bugzilla should apply:

https://bugzilla.redhat.com/show_bug.cgi?id=1395366

**Comment by yaronct** [comment]
@Claghorn: Do u mind openning a pull request here?

**Comment by Claghorn** [comment]
I'm afraid I have no idea what a pull request is, and visiting the pull request link didn't provide any enlightenment.

**Comment by yaronct** [comment]
@Claghorn: Pull requests are the standard way to contribute to a project in github. 1st, u have to learn git, which is a VCS (version control system). Then u make the changes locally on your computer, then open a pull request on github, which means a request to merge your changes into the main libepoxy development branch. All that isn't strictly necessary, however it makes everything easier, is highly recommended, and will increase the chances your changes will b approved, as well as speed up the process. You can read [here](https://help.github.com/articles/about-pull-requests/), but again, you'll 1st have to learn git. If u choose that path, all that will definitely take time, however it'll help u a lot with contributing to projects in general.

**Comment by ebassi** [comment]
The issue has been fixed in master by commit a15a92c2cbe0a8f45a1ff6258b22957c17c7118e — now Epoxy won't try to access `NULL` strings.

Thanks for your bug report!
